wgpu/base.py

```diff
@@ -327,17 +327,18 @@ def create_buffer_with_data(self, *, label="", data, usage: "GPUBufferUsageFlags
         """
         raise NotImplementedError()
 
+    # FIXME: was create_texture(self, *, label="", size: "structs.Extent3D", mip_level_count: int = 1, sample_count: int = 1, dimension: "enums.TextureDimension" = "2d", format: "enums.TextureFormat", usage: "flags.TextureUsage"):
     # IDL: GPUTexture createTexture(GPUTextureDescriptor descriptor);
     def create_texture(
         self,
         *,
         label="",
-        size: "GPUExtent3D",
-        mip_level_count: "GPUIntegerCoordinate" = 1,
-        sample_count: "GPUSize32" = 1,
-        dimension: "GPUTextureDimension" = "2d",
-        format: "GPUTextureFormat",
-        usage: "GPUTextureUsageFlags",
+        size: "list(int) or structs.Extent3D",
```

Comment:
is this a valid python type expression?

Reply:
Good one. It won't run, but at least flake8 considers it good enough.  Maybe we should do the `List(int)` annotation.

Reply:
... and is there a Uninion type annotation thing?

Reply:
see https://docs.python.org/3/library/typing.html :)

In this case you would write `Union[List[int], structs.Extent3D]` I think

There's been some changes in python 3.9 too regarding this subject, it's kind of a mess

Reply:
Flake8 only does partial parsing as well. If you really want to do type checking you need to use something like mypy or one of the more modern (read faster) alternatives, but it's all very immature and rabbit-holey... I wouldn't recommend it

Reply:
The important thing is that the annotation contains some more or less useful info for the user, and that flake8 accepts it :)

I'll implement the Union and List tho.

Reply:
Done. Still wrapped in a string, otherwise things get ugly in `structs.py`, but it looks very much like mypy code now :)